### ISSUE TITLE
Different Dashboards for Different Users

### ISSUE THREAD

**Issue by kevinmonisit** [HackRU/frontendv2#3]
As a hacker/admin/organizer, I want to see a dashboard that's meant for me when I go to `<base URL>/dashboard`.

<img width="1473" alt="image" src="https://github.com/HackRU/frontendv2/assets/7038712/dd08eb32-fae8-4551-be79-5c8b15864e5f">

What does this mean to you? What's the best way of implementing this? Perhaps there's some sort of state that determines whats on the screen based on user role?

 We can discuss in the comments of this Issue if you want. Most important thing is to make this very very minimal for now

**Comment by kevinmonisit** [comment]
In this discussions, we might have to discuss with Andrew to see how we can have some common area for abstracting the API. Currently, an abstraction lives in `/app/lib/data.ts`, which allows for **React Server Components** which means the fetching is done on the edge/CDNs. 

If you'd like to learn more specifics about that, you can check out https://nextjs.org/learn/dashboard-app/fetching-data.

In other words, data.ts will sorta be like the profile.js? I will cross post this to the other issue regarding mocking profile.js
